### PR TITLE
[RF][ROOT-9816] Improve RooAbsCollection

### PR DESCRIPTION
**Pull request by hageboeck**
Replace the RooLinkedList in the central collections of RooFit (RooArgSet, RooArgList) with a std::vector. This speeds up the tutorials of RooFit/RooStats by 10 to 20 %.
The interface with respect to iterators and size() now resembles an STL container. When iterating on the updated collections, only begin(), end() and range-based for loops should be used. The legacy RooFit iterators are still supported, but should not be used, since they are slower. They were flagged with the `R__SUGGEST_ALTERNATIVE` macro.

**For Reviewers**
Reviews are most appreciated in `RooAbsCollection.h` (interfaces) and maybe `RooAbsCollection.cxx`. This is the central change.
All other changes are mostly places where old iterators have been replaced by new iterators to make RooFit faster.

### CODE REVIEW

**Comment by hageboeck** [comment]
@phsft-bot build just on ROOT-performance-centos7-multicore/default

File: roofit/roofitcore/inc/RooAbsAnaConvPdf.h [hunk 1]
```diff
@@ -96,7 +96,7 @@ class RooAbsAnaConvPdf : public RooAbsPdf {
   RooListProxy _convSet  ;             //  Set of (resModel (x) basisFunc) convolution objects
   RooArgList _basisList ;              //!  List of created basis functions
   mutable RooArgSet* _convNormSet ;    //!  Subset of last normalization that applies to convolutions
-  mutable TIterator* _convSetIter ;    //! Iterator over _convNormSet
+//  mutable TIterator* _convSetIter ;    //! Iterator over _convNormSet
```
Comment (amadio) [inline roofit/roofitcore/inc/RooAbsAnaConvPdf.h]: Why not just remove the line?

Comment (hageboeck) [reply roofit/roofitcore/inc/RooAbsAnaConvPdf.h]: No reason. Just overlooked it.

**Comment by hageboeck** [comment]
@phsft-bot build

**Pull request merged**